### PR TITLE
feat:add runtime to easy use for createOrCreate

### PR DESCRIPTION
**Pull request by jiuker**
feat:add runtime to easy use for createOrCreate

### CODE REVIEW

**Comment by pjuarezd** [comment]
needs  a little `go mod tidy`

File: pkg/runtime/tools.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package runtime
```
Comment (harshavardhana) [inline pkg/runtime/tools.go:1]: Please add the missing license header, borrow from other files.

File: pkg/runtime/sync.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
+package runtime
```
Comment (harshavardhana) [inline pkg/runtime/sync.go:1]: license header misssing.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by pjuarezd — APPROVED** [review]

**Review by pjuarezd — APPROVED** [review]

**Comment by jiuker** [comment]
@harshavardhana @shtripat  PTAL

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**